File: 20211121/gql-server/models/task.rb
# frozen_string_literal: true

# Task model
class Task
  attr_accessor :id, :title, :status

  @tasks = []

  def initialize(id:, title:, status:)
    @id = id
    @title = title
    @status = status
  end

  def self.all
    @tasks
  end

  def self.add(task)
    @tasks << task
  end
end

File: 20220304/react/src/models/Task.ts
export default class Task {
  id: string;

  title: string;

  completed: boolean;

  constructor({ id, title, completed }: {
    id: string;
    title: string;
    completed?: boolean;
  }) {
    this.id = id;
    this.title = title;
    this.completed = !!completed;
  }

  toggle() {
    this.completed = !this.completed;
  }
}
